### ISSUE TITLE
mpy-cross optimisation errors on wiznet5k

### ISSUE THREAD

**Issue by bill88t** [adafruit/circuitpython#7363]
This is a followup issue regarding https://github.com/adafruit/Adafruit_CircuitPython_Wiznet5k/issues/67.
It is specifically about `8.x`.

TLDR; If you mpy-cross the Adafruit_circuitpython_wiznet5k module with any optimisation (so, `-O1` or greater), the entire module breaks apart.

- `.chip` detection is broken
- `.link_status` detection is broken
- dhcp is broken
- and basically every socket & request is also broken

I do not have the technical know-how regarding how mpy-cross works, but I am willing to take a look at it.
I will do some hex dumps and compare the `-O0` mpy to the `-O1` and determine the differences.
However I will not have the time for it in the next few weeks.
If someone in the meantime wishes to work on it, be my guest.

**Comment by jepler** [comment]
See discussion in the wiznet issue. It appears mpy-cross optimization levels are working as designed.
